src/sentry/rules/processing/buffer_processing.py

```diff
@@ -0,0 +1,160 @@
+
+    If the number is larger than the batch size, it will chunk the items and process them in batches.
+
+    The batches are replicated into a new redis hash with a unique filter (a uuid) to identify the batch.
+    We need to use a UUID because these batches can be created in multiple processes and we need to ensure
+    uniqueness across all of them for the centralized redis buffer. The batches are stored in redis because
+    we shouldn't pass objects that need to be pickled and 10k items could be problematic in the celery tasks
+    as arguments could be problematic. Finally, we can't use a pagination system on the data because
+    redis doesn't maintain the sort order of the hash keys.
+
+    `processing_task` will fetch the batch from redis and process the rules.
+    """
+    batch_size = options.get("delayed_processing.batch_size")
+    should_emit_logs = options.get("delayed_processing.emit_logs")
+    log_format = "{}.{}"
+
+    try:
+        processing_info = delayed_processing_registry.get(processing_type)(project_id)
+    except NoRegistrationExistsError:
+        logger.exception(log_format.format(processing_type, "no_registration"))
+        return
+
+    hash_args = processing_info.hash_args
+    task = processing_info.processing_task
+    filters: dict[str, BufferField] = asdict(hash_args.filters)
+
+    event_count = buffer.backend.get_hash_length(model=hash_args.model, field=filters)
+    metrics.incr(
+        f"{processing_type}.num_groups", tags={"num_groups": bucket_num_groups(event_count)}
+    )
+
+    if event_count < batch_size:
+        return task.delay(project_id)
+
+    if should_emit_logs:
+        logger.info(
+            log_format.format(processing_type, "process_large_batch"),
+            extra={"project_id": project_id, "count": event_count},
+        )
+
+    # if the dictionary is large, get the items and chunk them.
+    alertgroup_to_event_data = fetch_group_to_event_data(project_id, hash_args.model)
+
+    with metrics.timer(f"{processing_type}.process_batch.duration"):
+        items = iter(alertgroup_to_event_data.items())
+
+        while batch := dict(islice(items, batch_size)):
+            batch_key = str(uuid.uuid4())
+
+            buffer.backend.push_to_hash_bulk(
+                model=hash_args.model,
+                filters={**filters, "batch_key": batch_key},
+                data=batch,
+            )
+
+            # remove the batched items from the project alertgroup_to_event_data
+            buffer.backend.delete_hash(**asdict(hash_args), fields=list(batch.keys()))
+
+            task.delay(project_id, batch_key)
+
+
+def process_buffer() -> None:
+    fetch_time = datetime.now(tz=timezone.utc)
+    should_emit_logs = options.get("delayed_processing.emit_logs")
+
+    for processing_type, handler in delayed_processing_registry.registrations.items():
+        with metrics.timer(f"{processing_type}.process_all_conditions.duration"):
```

Comment:
should we have a consistent prefix on the metrics for `delayed_processing`?

Reply:
possibly, i can clean this up in a follow up